BlueprintUICommonControls/Sources/AccessibilityFocus.swift

```diff
@@ -0,0 +1,117 @@
+import BlueprintUI
+import UIKit
+
+/// Enables VoiceOver focus to jump to the wrapped element
+/// either on appearance or via a trigger that can be manually fired.
+public struct AccessibilityFocus: Element {
+
+    /// The element that will have the focus.
+    public var wrapping: Element
+
+    /// The type of accessibility notification that will be triggered.
+    public var notificationType: UIAccessibility.Notification
+
+    /// A object that can be held on to by the caller to manually trigger a focus.
+    public var trigger: Trigger
+
+    // MARK: Element
+
+    public var content: ElementContent {
+        ElementContent(child: wrapping)
+    }
+
+    public func backingViewDescription(with context: ViewDescriptionContext) -> ViewDescription? {
+        View.describe { config in
+            config[\.isAccessibilityElement] = false
+            config.apply { view in
+                view.apply(model: self)
+            }
+        }
+    }
+}
+
+// MARK: Backing view
+
+extension AccessibilityFocus {
+    private final class View: UIView {
+
+        func apply(model: AccessibilityFocus) {
+            model.trigger.action = { [weak self] in
+                guard let self = self else { return }
+                UIAccessibility.post(notification: model.notificationType, argument: self)
+            }
+
+            /// We need to wait a moment before firing the focus, to ensure that
+            /// we are not competing with UIAccessibility for the focus.
+            DispatchQueue.main.asyncAfter(deadline: .now() + 0.25) {
```

Comment:
Fwiw this is generally 0.15 in Market. It kinda makes me think we need an a11y queue or something... I'm not sure there is actually one right answer here wrt. "override the in progress actions" vs "wait for them to finish"...

Reply:
I don't know that a queue even work for this as voiceover will always override our notifications in favor of system announcements and we don't have any insight into wether any announcements happen to be in progress when we post our notification. 

There are a limited number of ways we can gain any insight or control into the announcements, but none work are helpful this context. 

we can observe the `UIAccessibility.announcementDidFinishNotification`, but this only gets sent for `.announcement` notifications, which expect a string rather than an element as an argument and don't alter accessibility focus.

Similarly, we can use the attributed string attribute 
`UIAccessibilitySpeechAttributeQueueAnnouncement` to queue an announcement behind any in progress speech, but again this is only useful for `.announcement` notifications and requires a string rather than an element. 

iOS 17 expands this by adding the concept of  UIAccessibilityPriority which allows us to post an an announcement that is uninterruptible, but again this is only available for attributed strings and `.announcement` notifications

Reply:
Fair enough – I was mostly thinking out loud. It's too bad there's no better way to deal with this. Ah well!